cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunTestsTask.java

```diff
@@ -465,4 +462,46 @@ private static void writeToTestSuiteJson(Map<String, TestSuite> testSuiteMap, Pa
         }
     }
 
+    private String getClassPath(JBallerinaBackend jBallerinaBackend, Package currentPackage) {
+        List<Path> dependencies = new ArrayList<>();
+        JarResolver jarResolver = jBallerinaBackend.jarResolver();
+
+        for (ModuleId moduleId : currentPackage.moduleIds()) {
+            Module module = currentPackage.module(moduleId);
+            if (!module.testDocumentIds().isEmpty()) {
+                Collection<Path> jarFilePathsRequiredForTestExecution =
+                        jarResolver.getJarFilePathsRequiredForTestExecution(module.moduleName());
+                dependencies.addAll(jarFilePathsRequiredForTestExecution);
+            }
+        }
+        dependencies = dependencies.stream().distinct().collect(Collectors.toList());
+
+        List<Path> jarList = getJarList(jBallerinaBackend, currentPackage);
+        dependencies.removeAll(jarList);
+
+        StringJoiner classPath = new StringJoiner(File.pathSeparator);
+        dependencies.stream().map(Path::toString).forEach(classPath::add);
+        return classPath.toString();
+    }
+
+    private List<Path> getJarList(JBallerinaBackend jBallerinaBackend, Package currentPackage) {
```

Comment:
Let's rename this to reflect the exact jar list. e.g. `getCurrentPackageThinJars`

Reply:
Ill change this to `getExclusionPathList` since we are resolving the multiple jars